### PR TITLE
[Compute] Communitygallery changes

### PR DESCRIPTION
**Pull request by haagha**
## Description

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1188

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

**Comment by haagha** [comment]
@isra-fel do you know how to fix this error:
ERROR: Assembly Microsoft.Azure.Management.ResourceGraph was not included in the required assemblies field for module Az.Compute

**Comment by isra-fel** [comment]
@haagha basically by adding a new reference you need also add the assembly to the "RequiredAssemblies" part of the PSD1 file.
Another issue is you need to use the same version of the package as used in other modules, [2.1.0 in your case](https://cs.github.com/?q=repo%3AAzure%2Fazure-powershell%20Microsoft.Azure.Management.ResourceGraph%20language%3AXML&scopeName=All%20repos&scope=) -- this is because otherwise assemblies with different version will conflict, and will cause Compute and other modules unable to be loaded together in PowerShell.

We'd like to avoid assembly conflict as much as possible. So my question is, is it completely necessary to reference resource graph SDK? Is there a workaround like using ARM template?

**Comment by isra-fel** [comment]
Kindly ping @haagha

File: src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs [hunk 1]
```diff
@@ -157,6 +166,55 @@ public void SharedGalleryGet()
...
+            {
+                CommunityGallery result = CommunityGalleriesClient.Get(this.Location, this.GalleryPublicName);
+                var psObject = new PSCommunityGallery();
+                ComputeAutomationAutoMapperProfile.Mapper.Map<CommunityGallery, PSCommunityGallery>(result, psObject);
+                WriteObject(psObject);
+            }
+            else if(this.IsParameterBound(c=> c.Community))
+            {
+                //find out if its ok to create client locally
+                ResourceGraphClient rgClient = AzureSession.Instance.ClientFactory.CreateArmClient<ResourceGraphClient>(DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
```
Comment (haagha) [inline src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs:181]: @isra-fel is this ok to initiate rgClient like this?

Comment (isra-fel) [reply src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs:181]: Yes

File: src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs [hunk 2]
```diff
@@ -157,6 +166,55 @@ public void SharedGalleryGet()
...
+                //        //resultList.Add(pageItem);
+                //    }
+                //    nextPageLink = pageResult.NextPageLink;
+                //}
+                //var psObject = new List<PSSharedGalleryList>();
+                //foreach (var r in resultList)
+                //{
+                //    //psObject.Add(ComputeAutomationAutoMapperProfile.Mapper.Map<SharedGallery, PSSharedGalleryList>(r));
+                //}
+                //WriteObject(psObject);
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs:214]: Please remove the commented code.

File: src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs [hunk 3]
```diff
@@ -157,6 +166,55 @@ public void SharedGalleryGet()
...
+                request.Query = query;
+                QueryResponse response = rgClient.Resources(request);
+                Dictionary<string,string> output = new Dictionary<string, string>();
+
+                var data = JsonConvert.DeserializeObject<List<PSCommunityGallery>>(response.Data.ToString());
+                
+                //string token = response.SkipToken;
+
+                //request.Options.SkipToken = token;
+                //response = rgClient.Resources(request);
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/Gallery/GalleryGetMethod.cs:194]: Please remove the commented code.

File: src/Compute/Compute/ChangeLog.md [hunk 4]
```diff
@@ -29,7 +29,8 @@
 * Added `Zone` and `IntentVMSizeList` optional parameters to the cmdlet `New-AzProximityPlacementGroup`.
 
 ## Version 4.30.0
-* Added parameters `PackageFileName`, `ConfigFileName` for `New-AzGalleryApplicationVersion` 
+* Added parameters `PackageFileName`, `ConfigFileName` for `New-AzGalleryApplicationVersion`
+* Changes made to Gallery cmdlets for Community Galleries
```
Comment (wyunchi-ms) [inline src/Compute/Compute/ChangeLog.md:33]: Please move the change log to the `Upcoming Release` section and make it past tense.

File: src/Compute/Compute/Generated/Models/PSCommunityGallery.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/Models/PSCommunityGallery.cs:1]: Please add the license at the head of the source file.

File: src/Compute/Compute/Generated/Models/PSCommunityGalleryImage.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/Models/PSCommunityGalleryImage.cs:1]: Please add the license at the head of the source file.

File: src/Compute/Compute/Generated/Models/PSCommunityGalleryImageVersion.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/Models/PSCommunityGalleryImageVersion.cs:1]: Please add the license at the head of the source file.

File: src/Compute/Compute/help/Get-AzGalleryImageVersion.md [hunk 8]
```diff
@@ -5,219 +5,228 @@ online version: https://docs.microsoft.com/powershell/module/az.compute/get-azga
 schema: 2.0.0
 ---
 
-# Get-AzGalleryImageVersion
+# Get-AzGalleryImageDefinition
```
Comment (wyunchi-ms) [inline src/Compute/Compute/help/Get-AzGalleryImageVersion.md:8]: Please revert this help file.

**Review by wyunchi-ms — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**